### ISSUE TITLE
JAWS announces all the buttons even if focus is on one button when there are multiple Buttons inside a Table cell (Focus Mode)

### ISSUE THREAD

**Issue by kevin-dhaduk** [FreedomScientific/VFO-standards-support#693]
### Summary

> If there are multiple buttons inside a Table cell, JAWS announces all the buttons when focus reaches to any one of them. This happens when focus arrives using Tab (Focus Mode).
> **Note**: Issue is not present with Browse Mode (navigating with Arrow keys).
>
> Example:
> 1. Suppose there are 3 buttons in a Table Cell called Edit, Hide, Delete buttons
> 2. Tab through the buttons
> 3. JAWS announces all 3 buttons even if the focus is on one of them. (If focus is on Delete button, JAWS announces "Edit button Hide button Delete button, to activate press enter")

### Expected result

> JAWS should announce just the Button that gets focused.

### Actual result

> JAWS is announcing all the buttons even if focus is on one button inside a Table cell. User may get confused where is the focus in real.

### Example

> 1. Go to https://codepen.io/kevindhaduk/pen/xxJLNxv
> 2. Table through the buttons inside the table

#### JAWS version and build number

>Tested on each of the following JAWS builds:
>>2022.2211.7
>>2023.2212.23
 
#### Operating System and version

> Windows 10 - Version 21H1 (OS Build 19043.928)

#### Browser and version:

> Chrome 109.0.5414.74
> Microsoft Edge 108.0.1462.76

#### Attachments

![JAWS - Multiple buttons inside a Table cell](https://user-images.githubusercontent.com/86653322/212735741-5214392f-e14e-40da-99fe-d4f994b3f59a.png)

**Comment by JAWS-test** [comment]
I have heard about the bug before, but I can't recreate it on my end. I tested it with Chrome and Egde + JAWS 2021, 2022 and 2023 and the bug never occurs when I navigate the table with TAB.

**Comment by stevefaulkner** [comment]
I have brought this issue up with the software team. I tested it with latest JAWS and Firefox/Chrome and could replicate.
@JAWS-test odd that it doesn't happen for you. Are you using default settings?

**Comment by JAWS-test** [comment]
> Are you using default settings?

Yes

**Comment by stevefaulkner** [comment]
Tested with JAWS June 2023 update (2023.2306.38) and could not reproduce. Closing as fixed

**Comment by robmc** [comment]
I'm experiencing this with JAWS 2023.2023.144.

**Comment by stevefaulkner** [comment]
@robmc please check with latest(2024) version and see if you can still reproduce

**Comment by kevin-dhaduk** [comment]
@stevefaulkner @JAWS-test 
The original issue is fixed in the latest version. Thanks for the fix.

Facing a similar issue with Checkbox. Please let me know if we already have any open issue for this.

> It seems like when there is a checkbox inside a table cell with some other content, JAWS presumes the whole cell content as its label

**Example:**
> Go to https://codepen.io/kevindhaduk/pen/vENpBpa
> I have added multiple types of cases like a checkbox with plain text, multiple checkboxes, a checkbox with other type of control
> Tab through the checkboxes inside the table

**Actual result:**
> JAWS reads everything inside the respective cell presuming the whole content as its label

**Expected result:**
> JAWS should read only the respective checkbox label when it gets focused

**JAWS version:**
> JAWS 2025.2506.170 - July 2025

**Attachments:**
<img width="1920" height="836" alt="Image" src="https://github.com/user-attachments/assets/97139b63-f9eb-40c7-9e7f-dd71ac009e87" />

**Comment by kevin-dhaduk** [comment]
@stevefaulkner @JAWS-test 
Please review the previous comment.

**Comment by JAWS-test** [comment]
I can confirm the bug. In my opinion, it is serious. It does not occur with NVDA. It is also not a browser issue; it is definitely a JAWS bug. It would probably be a good idea to open a new issue here for this.

The problem only occurs with TAB navigation. In the element overview or when reading linearly and using quick navigation (F, X), the checkboxes are output correctly.

**Comment by scoslow** [comment]
Currently having this issue with JAWS version 2025.2508.120

When a user navigates to a checkbox element within a row. All the control elements within that row are announced.  This only happens when the user, uses TAB/SHIFT + TAB. Using the arrow key does not produce the same results. 

Tested this in NVDA and can confirm I do not have the same issue.

<img width="1448" height="631" alt="Image" src="https://github.com/user-attachments/assets/4fb123b4-babb-4bd1-a606-8b3bf11cf94a" />

**Comment by JAWS-test** [comment]
@kevin-dhaduk Since the issue with JAWS 2026 persists, it’s essential to open a new ticket. This ticket is closed, so nothing will happen. Can you open it and refer to https://codepen.io/kevindhaduk/pen/vENpBpa?
